ui/pages/asset/components/asset-v2.tsx

```diff
@@ -0,0 +1,368 @@
+  | {
+      type: AssetType.native;
+      /** Whether the symbol has been verified to match the chain */
+      isOriginalNativeSymbol: boolean;
+    }
+  | {
+      type: AssetType.token;
+      /** The token's contract address */
+      address: string;
+      /** The number of decimal places to move left when displaying balances */
+      decimals: number;
+      /** An array of token list sources the asset appears in, e.g. [1inch,Sushiswap]  */
+      aggregators?: [];
+    }
+) & {
+  /** The hexadecimal chain id */
+  chainId: `0x${string}`;
+  /** The asset's symbol, e.g. 'ETH' */
+  symbol: string;
+  /** The asset's name, e.g. 'Ethereum' */
+  name?: string;
+  /** A URL to the asset's image */
+  image: string;
+  /** The current price of 1 unit of the asset, in the user's fiat currency */
+  currentPrice?: number;
+  balance: {
+    /**
+     * A decimal representation of the balance before applying
+     * decimals e.g. '12300000000000000' for 0.0123 ETH
+     */
+    value: string;
+    /**
+     * A displayable representation of the balance after applying
+     * decimals e.g. '0.0123' for 12300000000000000 WEI
+     */
+    display: string;
+    /** The balance's localized value in fiat e.g. '$12.34' or '56,78 €' */
+    fiat?: string;
+  };
+};
+
+// A page representing a native or token asset
+const AssetV2 = ({
+  asset,
+  optionsButton,
+}: {
+  asset: Asset;
+  optionsButton: React.ReactNode;
+}) => {
+  const t = useI18nContext();
+
+  const currency = useSelector(getCurrentCurrency);
+  const isSwapsChain = useSelector(getIsSwapsChain);
+  const isBuyableChain = useSelector(getIsBuyableChain);
+  const isBridgeChain = useSelector(getIsBridgeChain);
+
+  const [marketData, setMarketData] = useState<any>();
+  const headerRef = useRef<{ setBalanceOpacity: (o: number) => void }>(null);
+
+  const { chainId, type, symbol, name, image, balance } = asset;
+
+  const address =
+    type === AssetType.token
+      ? asset.address
+      : '0x0000000000000000000000000000000000000000';
+
+  if (chainSupportsPricing(chainId)) {
+    useEffect(() => {
+      setMarketData(undefined);
+
+      fetchWithCache({
+        url: `https://price-api.metafi.codefi.network/v2/chains/${chainId}/spot-prices/?includeMarketData=true&tokenAddresses=${address}&vsCurrency=${currency}`,
+        cacheOptions: { cacheRefreshTime: 10 * MINUTE },
+        functionName: 'GetAssetMarketData',
+      })
+        .catch(() => null)
+        .then((data) => setMarketData(data?.[address.toLowerCase()]));
+    }, [chainId, address, currency]);
+  }
+
+  return (
+    <Box
+      display={Display.Flex}
+      flexDirection={FlexDirection.Column}
+      className="asset__wrapper"
+    >
+      <AssetHeader
+        ref={headerRef}
+        image={image}
+        balance={balance}
+        optionsButton={optionsButton}
+      />
+      <Box className="main-container asset__container">
+        <Box
+          marginLeft={'auto'}
+          marginRight={'auto'}
```

Comment:
These don't need `{}`

Reply:
Fixed in https://github.com/MetaMask/metamask-extension/pull/22508/commits/b8d898ae385d72eb86eb05ae9024260f6187dc21